### PR TITLE
Unpin jsonnet dependencies

### PR DESCRIPTION
**Pull request by paulfantom**
I am unpinning jsonnet dependencies as `release-0.8` is out. This is also bringing in recent changes from kubernetes-mixin, prometheus-operator mixin, and kube-state-metics mixin.

/cc @prometheus-operator/kube-prometheus-reviewers @dgrisonnet

### CODE REVIEW

**Comment by paulfantom** [comment]
Seems like tests are failing due to changes in thanos mixin. @kakkoyun do you have ideas what changed?

**Comment by paulfantom** [comment]
Added `dimensions` to thanos mixin config so it may pass unit test.

File: jsonnet/kube-prometheus/components/prometheus.libsonnet [hunk 1]
```diff
@@ -53,8 +53,10 @@ function(params) {
           (import 'github.com/kubernetes-monitoring/kubernetes-mixin/alerts/add-runbook-links.libsonnet') + (
     if p._config.thanos != {} then
       (import 'github.com/thanos-io/thanos/mixin/alerts/sidecar.libsonnet') + {
+        targetGroups: {},
         sidecar: {
           selector: p._config.mixin._config.thanosSelector,
+          dimensions: std.join(', ', ['job', 'instance']),
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/components/prometheus.libsonnet:59]: Those are necessary to allow usage of latest thanos mixin.

**Review by dgrisonnet — APPROVED** [review]
LGTM

**Pull request merged**